src/vm/dllimportcallback.cpp

```diff
@@ -1416,6 +1416,7 @@ VOID UMThunkMarshInfo::RunTimeInit()
     {
         // caller pop
         m_cbRetPop = 0;
+        m_cbActualArgSize += numRegistersUsed * STACK_ELEM_SIZE;
```

Comment:
The RefIntTest tests look like a simple enough to reason about. I assume the failing test was the DoCallBack_MarshalRefInt_Stdcall, right? Could you please look at what is the m_cbActualArgSize and numRegistersUsed before the if / else when this test is executed?